### ISSUE TITLE
How to implement multi-GPU training?

### ISSUE THREAD

**Issue by jianyuheng** [Qualcomm-AI-research/lr-qat#1]

**Comment by jianyuheng** [comment]
@oderooij-qc Unable to reproduce the results on llama-7b. Could the training script be missing some hyperparameter settings, such as lora_alpha?

**Comment by y-bond** [comment]
Hey @jianyuheng,
Can you please share your exact setup and configuration of LLaMA-7B for which you are unable to reproduce? And what results do you obtain? In preparation of the code release we tested that we get exactly same results as stated in the paper.
 
Here a few questions/suggestions that might help you in debugging the issue:
* Have you tried the example script for Tiny-LLaMA (https://github.com/Qualcomm-AI-research/lr-qat?tab=readme-ov-file#example-run-command-tiny-llama) and did you manage to get close results?
* All the hyper-parameters which are not explicitly listed in the run command are set to the same defaults we used for our experiments (including lora_alpha which is set to 1)
* We did all experiment on a single-GPU. I assume for multi-GPU you need to change the accelerate config (num_machines & num_processes) & adjust learning rate and maybe gradient clipping to get the same results as on a single machine.

**Comment by markusnagel-qc** [comment]
Closed due to inactivity and missing information.
